ingestion/src/metadata/ingestion/api/source.py

```diff
@@ -69,3 +69,10 @@ def get_status(self) -> SourceStatus:
     @abstractmethod
     def test_connection(self) -> None:
         pass
+
+
+@dataclass  # type: ignore[misc]
```

Comment:
IDK why we have all sources decorated with this. If it is not needed, we can clean it up